application/core/src/main/java/com/github/pieter_groenendijk/model/fine/FineType.java

```diff
@@ -0,0 +1,43 @@
+package com.github.pieter_groenendijk.model.fine;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.Min;
+
+/**
+ * This is not an enum since it's very important that it's always consistent. We don't want stored fines for which
+ * the fine type is gone.
+ * By using a table, updates to this entity will be done on database level, (mostly) ensuring one source of truth.
+ */
+@Entity
+@Table(name = "FineType")
+public class FineType {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "fineTypeId")
+    private Long fineTypeId;
+
+    @Column(
+        name = "title",
+        length = 50,
+        unique = true,
+        nullable = false
+    )
+    private String title;
+
+    @Column(
+        name = "amountInCents",
+        nullable = false
+    )
+    @Min(0)
```

Comment:
@Lderksen27 Is het normaal om zulke validatie regels zowel op entity level, als op db level, te definiëren?

Reply:
Ik zou zeggen van wel. Het is althans prima te verdedigen.

Door de validatie op databaseniveau te hebben dwing je af dat wanneer er in de toekomst andere systemen met de database gaan praten, deze ook gedwongen zijn zich aan deze regels te houden. Zou het hierom niet per se met echte business-logica doen, maar voor dit soort zaken zeker.

De validatie op de entity is dan nog steeds nuttig om geen overbodig verkeer richting database te creëren. Ook makkelijker met de errorafhandeling.